src/coreclr/vm/vars.hpp

```diff
@@ -548,27 +548,9 @@ inline bool CORDebuggerAttached()
 
 
 //
-// Define stuff for precedence between profiling and debugging
-// flags that can both be set.
+// Note: Profiler now influences dwDebuggerBits initialization
 //
 
-#if defined(PROFILING_SUPPORTED) || defined(PROFILING_SUPPORTED_DATA)
-
-#ifdef DEBUGGING_SUPPORTED
-
-#define CORDisableJITOptimizations(dwDebuggerBits)        \
```

Comment:
`CORDisableJITOptimizations` is used in number of places. Do those other places need to be update too now that it does not include the profiler flags?

Should we delete `CORDisableJITOptimizations` macro and replace its uses with `CORDebuggerAllowJITOpts`? Both macros are doing the same thing now.

Reply:
Something I am wondering, is if there is a real case where `DEBUGGING_SUPPORTED` is unset while `defined(PROFILING_SUPPORTED) || defined(PROFILING_SUPPORTED_DATA)` is true. Because in that case there is a change of behavior: we unconditionally return false while in the past, the profiler could disable JIT optimizations.

As the debug bits are always got from the module object, I wonder if it would make sense to remove the macro and have a member function on module instead - marked with inlining hints - that would look at debug flags + profiler de-optimization flag.
I actually made the profiler flag influence the debugBits initialization to satisfy that macro. It feels a bit wrong though as those bits are logically owned by the Debugger IIUC.

Reply:
> As the debug bits are always got from the module object, I wonder if it would make sense to remove the macro and have a member function on module instead

Sounds good to me.

Reply:
While fixing that, I found a 2 instances of potentially wrong calls to the underlying "debugger-only" macro `CORDebuggerAllowJITOpts` - I think they should take into account profiler flag as well: 
- https://github.com/simonferquel/runtime/blob/d27ab2599124d9018704f5c1c8a22d75fbbea474/src/coreclr/debug/ee/debugger.inl?plain=1#L87C1-L87C49
- https://github.com/simonferquel/runtime/blob/d27ab2599124d9018704f5c1c8a22d75fbbea474/src/coreclr/debug/ee/debugger.inl?plain=1#L65C1-L65C64

Should I update those call sites as well ?

Reply:
> For the m_fHasOptimizedCode, it doesn't seem to be actually used anywhere, is it just for inspection to tell if the module is optimized? Could we add a note for its utility or remove it?

Agreed, that appears to be dead code. The field and its initialization can be deleted.
For the other one (DebuggerModule::HasAnyOptimizedCode()), yes I would update it.

Reply:
If the debugger macro should never be used outside of the method I introduced, shouldn't I move the macro code directly inside that method?

Reply:
Also, there is no "Profiler only" macro or method accessible outside of Module class, this is encapsulated in `AreJITOptimizationsDisabled` that checks both debugger and profiler bits.

Reply:
> shouldn't I move the macro code directly inside that method?

I like how the PR has it now with CORDisableJITOptimizations removed but CORDebuggerAllowJITOpts still existing. CORDebuggerAllowJITOpts also refers to g_CORDebuggerControlFlags and there are several other macros right next to it that show how other parts of g_CORDebuggerControlFlags are interpretted. I think that is useful.

@mdh1418 @jkotas - if you have a strong opinion otherwise do say so. I rarely have strong opinions on these subjective code refactoring choices but I want to give Simon clarity so this doesn't drag out.

Reply:
>  like how the PR has it now with CORDisableJITOptimizations removed but CORDebuggerAllowJITOpts still existing. CORDebuggerAllowJITOpts also refers to g_CORDebuggerControlFlags and there are several other macros right next to it that show how other parts of g_CORDebuggerControlFlags are interpretted. I think that is useful.

I actually removed it in my last batch of changes, but I can revert

Reply:
If looks fine to me both with and without `CORDebuggerAllowJITOpts` macro. If you are going to delete `CORDebuggerAllowJITOpts` macro, you may want to also delete `CORDebuggerEnCMode` since that macro looks very one-off with `CORDebuggerAllowJITOpts` deleted.

Reply:
Ill revert it and add a comment about this only looking at debugger bits and that in most cases "AreJITOptimizationsDisabled" should be used